From 6430d2020768f31fb351314f75e5c15601289f07 Mon Sep 17 00:00:00 2001
From: Salvo Giangreco <giangrecosalvo9@gmail.com>
Date: Mon, 6 Oct 2025 14:35:37 +0200
Subject: [PATCH] Allow custom PackageBlockListPolicy

UN1CA: retargeted using the available `.orig`/`.rej` smali artifacts for this target.

UN1CA: `ScanPackageUtils` no longer contains the old blocklist hunk on this base, so this patch only updates the remaining live call sites.

UN1CA: this patch is not complete! Please check unica/mods/blocklist/customize.sh for more info.
---
 .../server/pm/InstallPackageHelper.smali      | 12 ++---
 .../pm/install/PackageBlockListPolicy.smali   | 19 +-----
 .../server/pm/lifecycle/PmLifecycleImpl.smali | 49 -------------------
 3 files changed, 6 insertions(+), 75 deletions(-)

diff --git a/smali_classes2/com/android/server/pm/InstallPackageHelper.smali b/smali_classes2/com/android/server/pm/InstallPackageHelper.smali
index d8893e3df..1ac76a316 100644
--- a/smali_classes2/com/android/server/pm/InstallPackageHelper.smali
+++ b/smali_classes2/com/android/server/pm/InstallPackageHelper.smali
@@ -21808,20 +21808,16 @@
     :try_start_5
-    sget-object v2, Lcom/samsung/android/server/pm/install/PackageBlockListPolicy;->sIsRduDevice:Ljava/util/concurrent/atomic/AtomicBoolean;
-
-    invoke-virtual {v2}, Ljava/util/concurrent/atomic/AtomicBoolean;->get()Z
-
-    move-result v2
+    const/4 v2, 0x1
 
     if-nez v2, :cond_7
 
     goto :goto_3
 
     :cond_7
-    sget-object v2, Lcom/samsung/android/server/pm/install/PackageBlockListPolicy;->sLduBlocklist:Ljava/util/HashSet;
+    sget-object v2, Lcom/samsung/android/server/pm/install/PackageBlockListPolicy;->sBlocklist:Ljava/util/HashSet;
 
     if-nez v2, :cond_8
 
-    const-string v2, "/system/etc/ldu_blocklist.xml"
+    const-string v2, "/system/etc/unica_blocklist.xml"
 
     invoke-static {v2}, Lcom/samsung/android/server/pm/install/PmConfigParser;->parsePackages(Ljava/lang/String;)Ljava/util/List;
 
@@ -21831,7 +21827,7 @@
 
     invoke-direct {v3, v2}, Ljava/util/HashSet;-><init>(Ljava/util/Collection;)V
 
-    sput-object v3, Lcom/samsung/android/server/pm/install/PackageBlockListPolicy;->sLduBlocklist:Ljava/util/HashSet;
+    sput-object v3, Lcom/samsung/android/server/pm/install/PackageBlockListPolicy;->sBlocklist:Ljava/util/HashSet;
 
     :cond_8
     sget-boolean v2, Lcom/samsung/android/server/pm/install/PackageBlockListPolicy;->sBypassBlocklist:Z
@@ -21843,6 +21839,6 @@
     goto :goto_2
 
     :cond_9
-    sget-object v2, Lcom/samsung/android/server/pm/install/PackageBlockListPolicy;->sLduBlocklist:Ljava/util/HashSet;
+    sget-object v2, Lcom/samsung/android/server/pm/install/PackageBlockListPolicy;->sBlocklist:Ljava/util/HashSet;
 
     invoke-virtual {v2, v1}, Ljava/util/HashSet;->contains(Ljava/lang/Object;)Z
 
diff --git a/smali_classes2/com/samsung/android/server/pm/install/PackageBlockListPolicy.smali b/smali_classes2/com/samsung/android/server/pm/install/PackageBlockListPolicy.smali
index e6a229638..eeed61f8e 100644
--- a/smali_classes2/com/samsung/android/server/pm/install/PackageBlockListPolicy.smali
+++ b/smali_classes2/com/samsung/android/server/pm/install/PackageBlockListPolicy.smali
@@ -6,22 +6,4 @@
 # static fields
 .field public static sBypassBlocklist:Z = false
 
-.field public static final sIsRduDevice:Ljava/util/concurrent/atomic/AtomicBoolean;
-
-.field public static sLduBlocklist:Ljava/util/HashSet;
-
-
-# direct methods
-.method static constructor <clinit>()V
-    .locals 2
-
-    new-instance v0, Ljava/util/concurrent/atomic/AtomicBoolean;
-
-    const/4 v1, 0x0
-
-    invoke-direct {v0, v1}, Ljava/util/concurrent/atomic/AtomicBoolean;-><init>(Z)V
-
-    sput-object v0, Lcom/samsung/android/server/pm/install/PackageBlockListPolicy;->sIsRduDevice:Ljava/util/concurrent/atomic/AtomicBoolean;
-
-    return-void
-.end method
+.field public static sBlocklist:Ljava/util/HashSet;
diff --git a/smali_classes2/com/samsung/android/server/pm/lifecycle/PmLifecycleImpl.smali b/smali_classes2/com/samsung/android/server/pm/lifecycle/PmLifecycleImpl.smali
index 91512edd5..f87336b76 100644
--- a/smali_classes2/com/samsung/android/server/pm/lifecycle/PmLifecycleImpl.smali
+++ b/smali_classes2/com/samsung/android/server/pm/lifecycle/PmLifecycleImpl.smali
@@ -3228,46 +3228,4 @@
 
     move-result-object v0
 
-    sget-object v3, Lcom/samsung/android/server/pm/install/PackageBlockListPolicy;->sLduBlocklist:Ljava/util/HashSet;
-
-    invoke-virtual {v2}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
-
-    move-result-object v2
-
-    sget-object v3, Lcom/samsung/android/server/pm/install/PackageBlockListPolicy;->sIsRduDevice:Ljava/util/concurrent/atomic/AtomicBoolean;
-
-    const-string/jumbo v4, "shopdemo"
-
-    const/4 v5, 0x0
-
-    invoke-static {v2, v4, v5}, Landroid/provider/Settings$Secure;->getInt(Landroid/content/ContentResolver;Ljava/lang/String;I)I
-
-    move-result v6
-
-    const/4 v7, 0x1
-
-    if-ne v6, v7, :cond_18
-
-    const/4 v6, 0x1
-
-    goto :goto_13
-
-    :cond_18
-    move v6, v5
-
-    :goto_13
-    invoke-virtual {v3, v6}, Ljava/util/concurrent/atomic/AtomicBoolean;->set(Z)V
-
-    invoke-static {v4}, Landroid/provider/Settings$Secure;->getUriFor(Ljava/lang/String;)Landroid/net/Uri;
-
-    move-result-object v3
-
-    new-instance v4, Lcom/samsung/android/server/pm/install/PackageBlockListPolicy$1;
-
-    invoke-direct {v4, v0, v2}, Lcom/samsung/android/server/pm/install/PackageBlockListPolicy$1;-><init>(Landroid/os/Handler;Landroid/content/ContentResolver;)V
-
-    const/4 v6, -0x1
-
-    invoke-virtual {v2, v3, v5, v4, v6}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;I)V
-
     iget-object v0, v1, Lcom/samsung/android/server/pm/lifecycle/PmLifecycleImpl;->mATCommandReceiver:Lcom/samsung/android/server/pm/mm/MaintenanceModeManager$ATCommandReceiver;
 
-- 
2.51.0
